Description: add hardening flags to "guess" binaries.
 This doesn't change the resulting binary, but it
 avoids blhc warnings.
Origin: vendor
Forwarded: not-needed
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2022-12-03

--- a/Makefile.PL
+++ b/Makefile.PL
@@ -286,7 +286,7 @@
 sub compile_rlver {
     my ($base, $RLINC, $RLLIB, $defs, $lddflags, $libs, $quiet) = @_;
 
-    my $comp_cmd = "$Config{cc} $RLINC $Config{ccflags} $defs $base.c -o $base $RLLIB $lddflags $Config{ldflags} $libs";
+    my $comp_cmd = "$Config{cc} $RLINC $Config{ccflags} $ENV{CFLAGS} $ENV{CPPFLAGS} $ENV{LDFLAGS} $defs $base.c -o $base $RLLIB $lddflags $Config{ldflags} $libs";
     $comp_cmd .= " >/dev/null 2>&1" if $quiet && !$verbose;
     print $comp_cmd, "\n" if $verbose;
     system($comp_cmd);
@@ -367,7 +367,7 @@
     for my $symbol_set (@symbol_sets) {
         my $xdef = join " ", map "-D$_=$symbol_set->{$_}", sort keys %$symbol_set;
         # compile it
-        my $comp_cmd = "$Config{cc} $RLINC $Config{ccflags} $defs $xdef $frlmalloc -o rlmalloc $RLLIB $lddflags $Config{ldflags} $libs";
+        my $comp_cmd = "$Config{cc} $RLINC $Config{ccflags} $ENV{CFLAGS} $ENV{CPPFLAGS} $ENV{LDFLAGS} $defs $xdef $frlmalloc -o rlmalloc $RLLIB $lddflags $Config{ldflags} $libs";
         print $comp_cmd, "\n" if $verbose;
         return $xdef if (system($comp_cmd) == 0 && `./rlmalloc` =~ /^ok\s*$/); # maybe terminated by "\r\n"
     }
